src/Blazored.Typeahead/BlazoredTypeahead.razor.cs

```diff
@@ -11,26 +11,31 @@ namespace Blazored.Typeahead
     public class BlazoredTypeaheadBase<TItem> : ComponentBase, IDisposable
     {
         [Inject] IJSRuntime JSRuntime { get; set; }
-
         [Parameter] public string Placeholder { get; set; }
         [Parameter] public TItem Value { get; set; }
         [Parameter] public EventCallback<TItem> ValueChanged { get; set; }
-        [Parameter] public Func<string, Task<List<TItem>>> SearchMethod { get; set; }
+        [Parameter] public Func<string, Task<IEnumerable<TItem>>> SearchMethod { get; set; }
         [Parameter] public RenderFragment NotFoundTemplate { get; set; }
         [Parameter] public RenderFragment<TItem> ResultTemplate { get; set; }
         [Parameter] public RenderFragment<TItem> SelectedTemplate { get; set; }
         [Parameter] public RenderFragment FooterTemplate { get; set; }
         [Parameter] public int MinimumLength { get; set; } = 1;
         [Parameter] public int Debounce { get; set; } = 300;
+        [Parameter] public int MaximumSuggestions { get; set; } = 25;
 
         protected bool Searching { get; set; } = false;
-        protected bool EditMode { get; set; } = true;
-        protected List<TItem> SearchResults { get; set; } = new List<TItem>();
-
+        protected bool ShouldShowMenu { get; private set; } = false;
```

Comment:
I'm not sure of the naming here, `Shouldxxx` doesn't seem to read well... `if (ShouldShowMenu) { // do stuff }`. What about a change to `ShowingMenu`... `if (ShowingMenu) { // do stuff }`. This is very much personal a preference thing.

Reply:
Agreed, `IsShowing...` ?